src/pilot/assign_role.py

```diff
@@ -850,6 +850,16 @@ def assign_role(ip_mac_service_tag, node_uuid, role_index, os_volume_size_gb,
               'path': '/properties/capabilities'}]
     ironic_client.node.update(node_uuid, patch)
 
+    # Detecting BOSS Card and find the boss device size
+    lst_ctrls = drac_client.list_raid_controllers()
+    boss_disk = [ctrl.id for ctrl in lst_ctrls if ctrl.model == "BOSS-S1"]
+    lst_physical_disks = drac_client.list_physical_disks()
```

Comment:
I'm pretty sure there will be a problem when running assign_role against a node that does not have a BOSS card.  We need to detect that case and skip lines 856 through 862 in that case.

Reply:
I have handled this scenario.